packages/@aws-cdk/aws-events-targets/lib/aws-api.ts

```diff
@@ -127,3 +130,37 @@ function awsSdkToIamAction(service: string, action: string): string {
   const iamAction = action.charAt(0).toUpperCase() + action.slice(1);
   return `${iamService}:${iamAction}`;
 }
+
+const sdkDocsUrl = 'https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/index.html';
+
+/**
+ * Check that the given service and actions exist in the AWS SDK. If the service name is given in an incorrect case
+ * e.g. rds instead of RDS, the normalized service name is returned.
+ * @param service Service name
+ * @param action Action name
+ * @param apiVersion API version
+ * @returns Normalized service name
+ * @throws {Error} The given service and/or action don't exist
+ */
+function checkServiceAndActionExists(service: string, action: string, apiVersion?: string): string {
+  let sdkService: any;
+  let normalizedServiceName: string;
+
+  if ((SDK as any)[service]) {
+    sdkService = (SDK as any)[service];
+    normalizedServiceName = service;
+  } else if ((SDK as any)[service.toUpperCase()]) {
+    sdkService = (SDK as any)[service.toUpperCase()];
+    normalizedServiceName = service.toUpperCase();
+  } else {
+    throw new Error(`The service ${service} does not exist, check the list of available services from ${sdkDocsUrl}`);
+  }
+
+  const sdkServiceInstance = new sdkService(apiVersion && { apiVersion });
+  if (!sdkServiceInstance[action]) {
+    throw new Error(`The action ${action} for the service ${service} does not exist, check the list of available \
```

Comment:
The downside of throwing an error is that if our knowledge is incorrect or outdated (for whatever reason, probably because we didn't get the freshest build of the SDK and there's a shiny new API we don't know about yet), the CDK will completely fail to work.

There needs to either be an "I know what I'm doing" override, or this needs to turn into an `Annotations.of(this).addWarning()`.

Reply:
AwsApi is not a construct so using Annotations.of(this) doesn't work..

Reply:
How about adding the warning to the Lambda handler function?

Reply:
That'll do it, yeah.